internal/transport/http2_client.go

```diff
@@ -762,6 +775,7 @@ func (t *http2Client) Close() error {
 		}
 		t.statsHandler.HandleConn(t.ctx, connEnd)
 	}
+	t.onClose()
```

Comment:
if `t.Close()` is called before we assign `t.onClose`, this will panic.  Initialize `onClose` above to `func(){}` or check for `nil` here.

Aside: I'm surprised travis is green. (!)

Reply:
Yikes, good call. Done.

Travis is probably green because that race doesn't come up very often. (I only observed the prior race in maybe 1 out of 5 CLs previously).